Add haptic feedback

benonymity opened the pull request:
Adds haptic feedback to all buttons that perform a tangible action (play/pause, change setting, download book, etc.) I don't have access to a physical Android device, does the haptic feedback respond in the same way on those devices? This makes a ton of functions async but it doesn't seem to have any negative impacts that I noticed. Adds another dimensions in interacting with the app.

advplyr commented:
I made a haptics plugin a while ago but didn't actually put it to use.

You don't have to import the haptics at the top of each page/component if you use the plugin.

So for example instead of
`await Haptics.impact({ style: ImpactStyle.Medium })`
you can do
`await this.$hapticsImpactMedium()`

All those functions you will find in https://github.com/advplyr/audiobookshelf-app/blob/master/plugins/haptics.js

If you can make that update on those and also remove the semicolons. I'm also seeing semicolons in the haptics.js file that I must have missed when I was copying those methods in from the docs.

benonymity commented:
Oh, didn't realize you already implemented the plugin! Though I did wonder why CapacitorHapics was already in the Podfile... Just removed the capacitor imports using the existing plugin and removed the semicolons from the `haptics.js` file.

advplyr commented on pages/settings.vue:
```diff
@@ -47,6 +47,7 @@
 
 <script>
 import { Dialog } from '@capacitor/dialog'
+import { Haptics, ImpactStyle } from '@capacitor/haptics';
```
```suggestion
```

advplyr commented on pages/settings.vue:
```diff
@@ -149,6 +150,7 @@ export default {
       this.saveSettings()
     },
     async saveSettings() {
+      await Haptics.impact({ style: ImpactStyle.Medium });
```
```suggestion
      await this.$hapticsImpactMedium()
```

advplyr commented on components/home/BookshelfToolbar.vue:
```diff
@@ -8,7 +8,7 @@
         <p v-show="selectedSeriesName" class="ml-2 font-book pt-1">{{ selectedSeriesName }} ({{ totalEntities }})</p>
         <div class="flex-grow" />
-        <span v-if="page == 'library' || seriesBookPage" class="material-icons px-2" @click="bookshelfListView = !bookshelfListView">{{ !bookshelfListView ? 'view_list' : 'grid_view' }}</span>
+        <span v-if="page == 'library' || seriesBookPage" class="material-icons px-2" @click="changeView()">{{ !bookshelfListView ? 'view_list' : 'grid_view' }}</span>
```
```suggestion
        <span v-if="page == 'library' || seriesBookPage" class="material-icons px-2" @click="changeView">{{ !bookshelfListView ? 'view_list' : 'grid_view' }}</span>
```

advplyr commented:
Tested on Android and it works great. Thanks!
